processor/k8sattributesprocessor/processor.go

```diff
@@ -71,7 +71,11 @@ func (kp *kubernetesprocessor) Start(_ context.Context, _ component.Host) error
 		}
 	}
 	if !kp.passthroughMode {
-		go kp.kc.Start()
+		err := kp.kc.Start()
+		if err != nil {
+			kp.telemetrySettings.ReportStatus(component.NewFatalErrorEvent(err))
+			return nil
```

Comment:
We should return the error

Reply:
After I made all errors returned as is, the lifecycle unit tests fails with `unable to load k8s config, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined`. Should we make further changes to make it start normally with default configs and no environment variables?